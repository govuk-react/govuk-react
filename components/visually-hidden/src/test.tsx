import React from 'react';
import { mount } from 'enzyme';

import { VisuallyHiddenDocumented as VisuallyHidden } from '.';

describe('VisuallyHidden', () => {
  it('matches snapshot', () => {
    expect(mount(<VisuallyHidden>Example</VisuallyHidden>)).toMatchSnapshot('default render');
  });

  it('focusable version matches snapshot', () => {
    expect(mount(<VisuallyHidden focusable>Example</VisuallyHidden>)).toMatchSnapshot('focusable');
  });

  it('with important={false} matches snapshot', () => {
    expect(mount(<VisuallyHidden important={false}>Example</VisuallyHidden>)).toMatchSnapshot('not important');
  });
});
