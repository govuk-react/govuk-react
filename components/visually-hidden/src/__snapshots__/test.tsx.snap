// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisuallyHidden focusable version matches snapshot: focusable 1`] = `
.c0 {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: 0 !important;
  overflow: hidden !important;
  -webkit-clip: rect(0 0 0 0) !important;
  clip: rect(0 0 0 0) !important;
  -webkit-clip-path: inset(50%) !important;
  clip-path: inset(50%) !important;
  border: 0 !important;
  white-space: nowrap !important;
}

.c0:active,
.c0:focus {
  position: static !important;
  width: auto !important;
  height: auto !important;
  margin: inherit !important;
  overflow: visible !important;
  -webkit-clip: auto !important;
  clip: auto !important;
  -webkit-clip-path: none !important;
  clip-path: none !important;
  white-space: inherit !important;
}

<VisuallyHiddenDocumented
  focusable={true}
  important={true}
>
  <src__VisuallyHidden
    focusable={true}
    important={true}
  >
    <span
      className="c0"
      tabIndex="0"
    >
      Example
    </span>
  </src__VisuallyHidden>
</VisuallyHiddenDocumented>
`;

exports[`VisuallyHidden matches snapshot: default render 1`] = `
.c0 {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  margin: 0 !important;
  overflow: hidden !important;
  -webkit-clip: rect(0 0 0 0) !important;
  clip: rect(0 0 0 0) !important;
  -webkit-clip-path: inset(50%) !important;
  clip-path: inset(50%) !important;
  border: 0 !important;
  white-space: nowrap !important;
  padding: 0 !important;
}

<VisuallyHiddenDocumented
  focusable={false}
  important={true}
>
  <src__VisuallyHidden
    focusable={false}
    important={true}
  >
    <span
      className="c0"
    >
      Example
    </span>
  </src__VisuallyHidden>
</VisuallyHiddenDocumented>
`;

exports[`VisuallyHidden with important={false} matches snapshot: not important 1`] = `
.c0 {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: 0;
  overflow: hidden;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  border: 0;
  white-space: nowrap;
  padding: 0;
}

<VisuallyHiddenDocumented
  focusable={false}
  important={false}
>
  <src__VisuallyHidden
    focusable={false}
    important={false}
  >
    <span
      className="c0"
    >
      Example
    </span>
  </src__VisuallyHidden>
</VisuallyHiddenDocumented>
`;
